datastore/uncached.go

```diff
@@ -137,3 +142,37 @@ func (s *DataStore) GetAllConfigMaps() (runtime.Object, error) {
 func (s *DataStore) GetAllVolumeAttachments() (runtime.Object, error) {
 	return s.kubeClient.StorageV1().VolumeAttachments().List(context.TODO(), metav1.ListOptions{})
 }
+
+// FindARandomReadyNode return a random ready node.
+// Return error if there is no ready node
+func FindARandomReadyNode(ei *longhorn.EngineImage, nodeList *longhorn.NodeList) (string, error) {
+	if ei.Status.State != longhorn.EngineImageStateDeployed && ei.Status.State != longhorn.EngineImageStateDeploying {
```

Comment:
Do you know why we don't check `ei.Status.State != longhorn.EngineImageStateDeployed` ? because I don't see the below logic to check if the node really has the image ready or not. 

P.S. I know this is the previous implementation.

Reply:
I agree with @innobead we should check to see if the node is ready and has engine image deployed. Similar to the logic here [ListReadyNodesWithReadyEngineImage](https://github.com/longhorn/longhorn-manager/blob/44b41a6d41886e3ff249c5375697ae8156a7194b/datastore/longhorn.go#L1913)

Reply:
Fixed.